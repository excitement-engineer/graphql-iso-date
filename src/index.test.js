/**
 * Copyright (c) 2016, Dirk-Jan Rutten
 * All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

var Kind = require('graphql/language').Kind

import GraphQLDate from '../'

describe('GraphQLDate', () => {
  describe('serialization test', () => {
    it('serializes javascript dates', () => {
      expect(GraphQLDate.serialize(new Date(2012, 10, 10))).toEqual('2012-11-10')
      expect(GraphQLDate.serialize(null)).toEqual(null)
      expect(() => GraphQLDate.serialize(5)).toThrowErrorMatchingSnapshot()
      expect(() => GraphQLDate.serialize('2010-11-10')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('value parsing test', () => {
    it('parses values', () => {
      expect(GraphQLDate.parseValue('2012-11-10').getTime()).toEqual(new Date(2012, 10, 10).getTime())
      expect(GraphQLDate.parseValue(null)).toEqual(null)
      expect(() => GraphQLDate.parseValue(5)).toThrowErrorMatchingSnapshot()
      expect(() => GraphQLDate.parseValue('2011-02-29')).toThrowErrorMatchingSnapshot()
      expect(() => GraphQLDate.parseValue('2011/02/29')).toThrowErrorMatchingSnapshot()
      expect(() => GraphQLDate.parseValue('bla')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('literal parsing test', () => {
    it('parses literals', () => {
      expect(
        GraphQLDate.parseLiteral({kind: Kind.STRING, value: '2012-11-10'}).getTime()
      ).toEqual(new Date(2012, 10, 10).getTime())
      expect(() => {
        GraphQLDate.parseLiteral({kind: Kind.STRING, value: '2011-02-29'})
      }).toThrowErrorMatchingSnapshot()
      expect(() => {
        GraphQLDate.parseLiteral({kind: Kind.STRING, value: 'bla'})
      }).toThrowErrorMatchingSnapshot()
      expect(() => {
        GraphQLDate.parseLiteral({kind: Kind.INT, value: 5})
      }).toThrowErrorMatchingSnapshot()
    })
  })
})
