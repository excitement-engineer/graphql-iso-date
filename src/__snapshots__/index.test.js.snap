exports[`GraphQLDate literal parsing test parses literals 1`] = `"Invalid date 2011-02-29, only accepts dates in format \'YYYY-MM-DD\'"`;

exports[`GraphQLDate literal parsing test parses literals 2`] = `"Invalid date bla, only accepts dates in format \'YYYY-MM-DD\'"`;

exports[`GraphQLDate literal parsing test parses literals 3`] = `"Can only parse strings to a Date but got kind IntValue"`;

exports[`GraphQLDate serialization test serializes javascript dates 1`] = `"Date must be serialized from a javascript Date instance but got object with type \'number\' and value 5"`;

exports[`GraphQLDate serialization test serializes javascript dates 2`] = `"Date must be serialized from a javascript Date instance but got object with type \'string\' and value 2010-11-10"`;

exports[`GraphQLDate value parsing test parses values 1`] = `"Value must be parsed from a String but got object with type \'number\' and value 5"`;

exports[`GraphQLDate value parsing test parses values 2`] = `"Value 2011-02-29 is not a valid date in the format YYYY-MM-DD"`;

exports[`GraphQLDate value parsing test parses values 3`] = `"Value 2011/02/29 is not a valid date in the format YYYY-MM-DD"`;

exports[`GraphQLDate value parsing test parses values 4`] = `"Value bla is not a valid date in the format YYYY-MM-DD"`;
